spec/support/dfe_sign_in_user_helper.rb

```diff
@@ -24,6 +24,10 @@ def invalid_dfe_sign_in_response
     }
   end
 
+  def when_dsi_fails
+    OmniAuth.config.mock_auth[:dfe] = :invalid_credentials
```

Comment:
Do we need to reset this after this a test case is run?

Reply:
Most specs reset this themselves by calling https://github.com/DFE-Digital/itt-mentor-services/blob/cv/dsi-failure/spec/support/dfe_sign_in_user_helper.rb#L9